[ZEPPELIN-1789] Make exported data file name more explicit

AhyoungRyu opened the pull request:
### What is this PR for?
Currently we can download table data set as `.csv` or `.tsv` format in Zeppelin note. But since the exported file name is always set as `data.csv`, it will be difficult to distinguish the each file if user tries to export it several times. So I changed the naming rule like below. 

1. if the paragraph has title e.g. `this is title`, then the exported csv file name will be 
`this is title_YYYY-MM-DD hh:mm:ss a.csv` .

2. if not, the file name will be `data_YYYY-MM-DD hh:mm:ss a.csv` .

Please see the short discussion in [ZEPPELIN-1789](https://issues.apache.org/jira/browse/ZEPPELIN-1789) for the more details. I just made this PR by following one person's opinion who raised this issue (_Ruslan Dautkhanov_ ). So if anyone has better idea, please feel free to share :)

### What type of PR is it?
Improvement

### What is the Jira issue?
[ZEPPELIN-1789](https://issues.apache.org/jira/browse/ZEPPELIN-1789)

### How should this be tested?
1. run Zeppelin web dev mode under `zeppelin-web` 

```
$ npm run dev
```

2. browse `http://localhost:9000` and go to Zeppelin-Spark tutorial note
3. click ![screen shot 2016-12-30 at 2 26 32 pm](https://cloud.githubusercontent.com/assets/10060731/21560080/fcf86cdc-ce9b-11e6-8fe3-a8ea0bd9141d.png) 

If the paragraph has title, then the exported file name will be `"paragraph_title_YYYY-MM-DD hh:mm:ss a".csv`, if not, will be `"data_YYYY-MM-DD hh:mm:ss a".csv`.

> The timestamp is not current time but paragraph finished time


### Screenshots (if appropriate)


### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

Leemoonsoo commented:
Thanks @AhyoungRyu for the improvement.

How about use [`dateFinished`](https://github.com/apache/zeppelin/blob/master/zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/Job.java#L73) instead of current time in the file name?

AhyoungRyu commented:
@Leemoonsoo Sounds good. Let me update then :)

AhyoungRyu commented:
@Leemoonsoo Addressed your comment in [d7d0087](https://github.com/apache/zeppelin/pull/1822/commits/d7d00872859d8d4f91e11875bbcd8e1012c8e2a6). Please check if you have time.

Leemoonsoo commented on zeppelin-web/src/app/notebook/paragraph/result/result.controller.js:
```diff
@@ -580,6 +580,9 @@ import zeppelin from '../../../zeppelin';
     $scope.exportToDSV = function(delimiter) {
       var dsv = '';
+      var dateFinished = moment(paragraph.dateFinished).format('YYYY-MM-DD hh:mm:ss A');
+      var exportedFileName = paragraph.title ? paragraph.title + '_' + dateFinished : 'Exported data_' + dateFinished;
```
nit: shell we make file name starts with lower case? 
`data_` instead of `Exported data_`, what do you think?

AhyoungRyu commented:
@Leemoonsoo I've just updated as you said and the PR description as well. Please take a look :)

Leemoonsoo commented:
@AhyoungRyu Thanks~ LGTM!

AhyoungRyu commented:
CI failure is not related this issue 
```

Results :

Failed tests: 
  NotebookTest.testAbortParagraphStatusOnInterpreterRestart:760 expected:<ABORT> but was:<RUNNING>

Tests run: 150, Failures: 1, Errors: 0, Skipped: 0
```
it's reported in [ZEPPELIN-1857](https://issues.apache.org/jira/browse/ZEPPELIN-1857).

Merge if there are no more comments
